### PR TITLE
[TRRFFDA-632] Limit user upload file types

### PR DESCRIPTION
**Pull request by alintreznai**

### CODE REVIEW

File: rdrf/rdrf/initial_data/blacklisted_mime_types.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+def load_data(**kwargs):
+    init_blacklisted_file_types()
+
+
+def init_blacklisted_file_types():
+
+    BlacklistedMimeType.objects.get_or_create(
+        mime_type='application/octet-stream',
+        defaults = {
+            'description': 'Executables files/Archives/Binary data'
```
Comment (sztamas) [inline rdrf/rdrf/initial_data/blacklisted_mime_types.py]: typo - ``Executable files``

File: rdrf/rdrf/initial_data/blacklisted_mime_types.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+
+def load_data(**kwargs):
+    init_blacklisted_file_types()
+
+
+def init_blacklisted_file_types():
+
+    BlacklistedMimeType.objects.get_or_create(
+        mime_type='application/octet-stream',
```
Comment (sztamas) [inline rdrf/rdrf/initial_data/blacklisted_mime_types.py]: Can you please do a bit of research about what other files are considered "dangerous"?
To test I started with going to ``/bin/`` uploading some executables. ``application/x-executable``, and ``text/x-shellscript`` were accepted, when they shouldn't and it raises the question what other mime-types we shouldn't allow for sure.
Worth investigating a bit...

File: rdrf/rdrf/forms/dynamic/fields.py [hunk 3]
```diff
@@ -37,7 +41,26 @@ def validate(self, value):
...
+    def validate(self, value):
+        if not value:
+            return super().validate(value)
+        __, ext = os.path.splitext(value._name)
+        mime_type = magic.from_buffer(value.file.read(2048), mime=True)
+        value.file.seek(0)
+
+        blacklisted_mime_type = self._find_blacklisted_mime_type(mime_type)
+        if blacklisted_mime_type:
+            raise ValidationError(f"{blacklisted_mime_type.description} file types aren't allowed to be uploaded into the system !")
```
Comment (sztamas) [inline rdrf/rdrf/forms/dynamic/fields.py]: Should be translated.

**Review by sztamas — CHANGES_REQUESTED** [review]

File: rdrf/rdrf/initial_data/blacklisted_mime_types.py (hunk 1 above)
Comment (alintreznai) [reply rdrf/rdrf/initial_data/blacklisted_mime_types.py]: Updated.

File: rdrf/rdrf/initial_data/blacklisted_mime_types.py (hunk 2 above)
Comment (alintreznai) [reply rdrf/rdrf/initial_data/blacklisted_mime_types.py]: Added more mime-types to the initial data script to cover more file types.

File: rdrf/rdrf/forms/dynamic/fields.py (hunk 3 above)
Comment (alintreznai) [reply rdrf/rdrf/forms/dynamic/fields.py]: Updated.

**Review by sztamas — APPROVED** [review]

**Pull request merged**